### PR TITLE
[WFLY-8624] Create test case for http-interface authentication secured by Elytron http-authentication-factory

### PR DESCRIPTION
**Pull request by olukas**
https://issues.jboss.org/browse/WFLY-8624
https://issues.jboss.org/browse/JBEAP-10485

### CODE REVIEW

**Review by bstansberry — DISMISSED** [review]
These kinds of tests need to be in WildFly Core not in WildFly.

Unless what's being tested requires use of features found in full WildFly, it should be tested in core.

**Comment by olukas** [comment]
@bstansberry My first idea was add this test case to wildfly-core. However this test case uses Wildfly Elytron subsystem. This subsystem is not part of wildfly-core instance which is built in WildFly Core project.
Hence this test is added to WildFly project.

**Comment by darranl** [comment]
+1 the subsystem is not currently available in core - but once it is we do have a batch of tests that should be moved to core which will reduce the coordination needed between the two.

**Comment by bstansberry** [comment]
At a minimum this should not use arquillian. WildFlyTestRunner can be used in manualmode tests in full but  arquillian is not available in core.

**Comment by olukas** [comment]
@bstansberry I tried to use WildflyTestRunner but there some problems:
I need to setup server for the same configuration for more single tests. In current Arquillian aproach I use @InSequence for it (which cannot be used with WildflyTestRunner). I do not want to configure the same configuration (which requires reload of server) before every single test (due to increasing execution time). I cannot use @BeforeClass because it must be static method which means I cannot use injected ServerController there.

I think using Arquillian is more suitable approach for this test case.

File: testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/mgmt/elytron/HttpMgmtInterfaceElytronAuthenticationTestCase.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    }
+
+    private ManagementClient getManagementClient() {
+        final ModelControllerClient client = TestSuiteEnvironment.getModelControllerClient();
+        return new ManagementClient(client, TestSuiteEnvironment.getServerAddress(),
+                TestSuiteEnvironment.getServerPort(), "http-remoting");
+    }
+
+    private URL createSimpleManagementOperationUrl() throws URISyntaxException, IOException {
+        return new URL("http://" + host + ":9990/management?operation=attribute&name=server-state");
```
Comment (kwart) [inline testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/mgmt/elytron/HttpMgmtInterfaceElytronAuthenticationTestCase.java]: Does this work with IPv6? The JavaDoc of `Utils.getSecondaryTestAddress` says the square brackets are stripped.

**Comment by kwart** [comment]
Just one possible issue, otherwise LGTM. 

@bstansberry IMO the `WildFlyTestRunner` usage would only add unnecessary complexity to this test. The Arquillian way looks proper to me in this case.

**Comment by olukas** [comment]
@kwart Fixed. Thanks!

**Review by kwart — APPROVED** [review]

**Review by kabir — APPROVED** [review]
Approving paperwork

**Comment by jamezp** [comment]
@olukas The `ServerController` can be injected statically and should be usable from a `@BeforeClass`. Using a `@BeforeClass` would be better than using `@InSeqeuence` IMO as well.

**Comment by bstansberry** [comment]
Never mind about WildFlyTestRunner. It's not currently usable for elytron testing.

**Comment by kwart** [comment]
@bstansberry gave a green light to this PR in the HipChat discussion (WildFly Elytron room), if I understand it correctly:

> [11:39 PM] Brian Stansberry: @JosefCacek go ahead and stick with ARQ
> [11:39 PM] Brian Stansberry: we'll have to rework it later
> [11:42 PM] Brian Stansberry: @jperkins WDYT about helping @KenWills and I out by adding elytron integration to WildFly Test Runner?

**Comment by kwart** [comment]
Test rebased to latest branches and new PRs were send directly to upstream/downstream projects. This one can be IMO closed.

**Comment by olukas** [comment]
Closed based on @kwart comment.

**Pull request closed without merging**